bin/user/mqtt.py

```diff
@@ -537,3 +559,52 @@ def process_record(self, record, dbm):
                 if res != mqtt.MQTT_ERR_SUCCESS:
                     logerr("publish failed for %s: %s" %
                            (tpc, mqtt.error_string(res)))
+
+    PERIODS = {
+        'day':lambda _time_ts:weeutil.weeutil.archiveDaySpan(_time_ts),
+        'yesterday':lambda _time_ts:weeutil.weeutil.archiveDaySpan(_time_ts,1,1),
+        'week':lambda _time_ts:weeutil.weeutil.archiveWeekSpan(_time_ts),
+        'month':lambda _time_ts:weeutil.weeutil.archiveMonthSpan(_time_ts),
+        'year':lambda _time_ts:weeutil.weeutil.archiveYearSpan(_time_ts)}
+
+    def get_record(self, record, dbmanager):
+        """Augment record data with additional data from the archive.
+        Should return results in the same units as the record and the database.
+
+        returns: A dictionary of weather values"""
+
+        # run parent class
+        _datadict = super(MQTTThread,self).get_record(record,dbmanager)
+
+        # actual time stamp
+        _time_ts = _datadict['dateTime']
+
+        # go through all calculations
+        for agg_obs in self.calculations:
+            try:
+                tag = self.calculations[agg_obs].split('.')
+                if len(tag)==3:
+                    # example: day.rain.sum
+                    # '$' at the beginning is possible but not necessary
+                    if tag[0][0]=='$': tag[0] = tag[0][1:]
+                    # time period (day, yesterday, week, month, year)
+                    if tag[0] not in MQTTThread.PERIODS: 
+                        raise ValueError("unknown time period '%s'" % tag[0])
+                    ts = MQTTThread.PERIODS[tag[0]](_time_ts)
+                    # If the observation type is in _datadict, calculate
+                    # the aggregation.
+                    # Note: It is no error, if the observation type is not
+                    #       in _datadict, as _datadict can be a LOOP packet
+                    #       that does not contain all the observation
+                    #       types.
+                    if tag[1] in _datadict:
+                        # get aggregate value
+                        __result = weewx.xtypes.get_aggregate(tag[1],ts,tag[2],dbmanager)
+                        # convert to unit system of _datadict
+                        _datadict[agg_obs] = weewx.units.convertStd(__result,_datadict['usUnits'])[0]
+                        # register name with unit group if necessary
+                        weewx.units.obs_group_dict.setdefault(agg_obs,__result[2])
+            except (LookupError,ValueError,TypeError,weewx.UnknownType,weewx.UnknownAggregation,weewx.CannotCalculate) as e:
+                logerr('%s = %s: error %s' % (obs,tag,e))
```

Comment:
Not a fan of these "fuck it, let's catch all" clauses. Not saying it is wrong in your particular case. Would you be able to separate them out?

Reply:
For me it is no use to send WeeWX to crash if there is some little problem with some little extension.

Reply:
Certainly, but not the point.
If you are aware that line 123 might raise a `LookupError`, handle it there. Please don't take this in a bad way.

Reply:
Stupid question: So you suggest to repeat all the error handling code (in this case one line, only) for all the places where errors can occur?

In this case I want to make sure, that the code of this function never ever crashes WeeWX. If I do not want to use that "catch all" clause, I would have to surround every single line of code by a `try:` - `except:` clause. Don' you think that is a lot of code and difficult to read?